src/codecs/gif.rs

```diff
@@ -58,9 +58,15 @@ impl<R: Read> GifDecoder<R> {
         let mut decoder = gif::DecodeOptions::new();
         decoder.set_color_output(ColorOutput::RGBA);
 
+        // We previously used `Limits::default()`, but they weren't actually enforced.
+        // Suddenly enabling enforcement while keeping default limits would break existing users.
+        // Therefore, set no limit at least until the next semver break, 0.25.x
+        let mut limits = Limits::default();
+        limits.max_alloc = None;
```

Comment:
I guess we forgot to follow up on https://github.com/image-rs/image/issues/1890, but we should continue the convention defined there of having the `new()` method initialize to `no_limits()` rather than `default()`

Reply:
I've updated the comments to reflect that and simplified the code a bit. This PR should be good to go.